plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2733,6 +2747,31 @@ private static Optional<String> find(Map<String, ColumnHandle> assignments, Proj
         return Optional.empty();
     }
 
+    @Override
+    public void validateScan(ConnectorSession session, ConnectorTableHandle handle)
+    {
+        DeltaLakeTableHandle deltaLakeTableHandle = (DeltaLakeTableHandle) handle;
+
+        // Skip for ANALYZE  as DeltaLake connector does not support analyze property 'partitions'
+        if (isQueryPartitionFilterRequired(session)) {
+            if (deltaLakeTableHandle.getAnalyzeHandle().isPresent()) {
```

Comment:
`ANALYZE` should fail only when the table is actually partitioned.

We need also the `!partitionColumns.isEmpty()` condition here.